buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -328,6 +328,11 @@ class BuildPlugin implements Plugin<Project> {
                     }
                 }
                 generatePomTask.configure({ Task t -> t.dependsOn = ['generatePomFileForShadowPublication'] } as Action<Task>)
+
+                // have to defer this until archivesBaseName is set
+                project.afterEvaluate {
+                    publication.artifactId = project.convention.getPlugin(BasePluginConvention).archivesBaseName
```

Comment:
it's unfortunate MavenPublication doesn't use Providers :/

Reply:
My thoughts exactly.